src/RenderWebGL.js

```diff
@@ -629,30 +656,46 @@ class RenderWebGL extends EventEmitter {
             gl.stencilFunc(gl.ALWAYS, 1, 1);
             gl.stencilOp(gl.KEEP, gl.KEEP, gl.REPLACE);
             gl.colorMask(false, false, false, false);
-            this._drawThese([drawableID], ShaderManager.DRAW_MODE.silhouette, projection);
+            this._drawThese([drawableID, ...candidateIDs], ShaderManager.DRAW_MODE.silhouette, projection);
 
             gl.stencilFunc(gl.EQUAL, 1, 1);
             gl.stencilOp(gl.KEEP, gl.KEEP, gl.KEEP);
             gl.colorMask(true, true, true, true);
 
-            this._drawThese(candidateIDs, ShaderManager.DRAW_MODE.silhouette, projection,
-                {idFilterFunc: testID => testID !== drawableID}
-            );
+            this._drawThese(candidateIDs.concat([drawableID]), ShaderManager.DRAW_MODE.silhouette, projection);
         } finally {
             gl.colorMask(true, true, true, true);
             gl.disable(gl.STENCIL_TEST);
         }
 
-        const pixels = new Uint8Array(Math.floor(bounds.width * bounds.height * 4));
-        gl.readPixels(0, 0, bounds.width, bounds.height, gl.RGBA, gl.UNSIGNED_BYTE, pixels);
+        const pixels = new Uint8Array(Math.floor((bounds.width+1) * (bounds.height+1) * 4));
+        gl.readPixels(0, 0, bounds.width+1, bounds.height+1, gl.RGBA, gl.UNSIGNED_BYTE, pixels);
```

Comment:
again -- bound rectangles, can we talk about this one?  Is it just me or does a box from 10-20 have 11 pixels width? assuming 10 and 20 are output